bitvm/src/signatures/public.rs

```diff
@@ -0,0 +1,838 @@
+
+        println!("result: {:?}", result);
+        assert!(result.success);
+    }
+
+    #[test]
+    fn secret_key_from_string() {
+        assert_eq!(
+            Wots16::secret_from_str("password").to_lower_hex_string(),
+            "37303631373337333737366637323634"
+        );
+    }
+
+    #[derive(Serialize, Deserialize)]
+    struct TestVector {
+        #[serde(
+            serialize_with = "serialize_bytes_hex",
+            deserialize_with = "deserialize_bytes_hex"
+        )]
+        message: Vec<u8>,
+        #[serde(
+            serialize_with = "serialize_bytes_hex",
+            deserialize_with = "deserialize_bytes_hex",
+            rename = "secretKey"
+        )]
+        secret_key: Vec<u8>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex",
+            rename = "publicKey"
+        )]
+        public_key: Vec<Vec<u8>>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        witness: Vec<Vec<u8>>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        signature: Vec<Vec<u8>>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        compact_witness: Vec<Vec<u8>>,
+        #[serde(
+            serialize_with = "serialize_byte_matrix_hex",
+            deserialize_with = "deserialize_byte_matrix_hex"
+        )]
+        compact_signature: Vec<Vec<u8>>,
+    }
+
+    fn serialize_bytes_hex<S>(bytes: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&bytes.to_lower_hex_string())
+    }
+
+    fn deserialize_bytes_hex<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Vec::<u8>::from_hex(&s).map_err(serde::de::Error::custom)
+    }
+
+    fn serialize_byte_matrix_hex<S>(
+        byte_matrix: &[Vec<u8>],
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(byte_matrix.len()))?;
+        for entry in byte_matrix {
+            seq.serialize_element(&entry.to_lower_hex_string())?;
+        }
+        seq.end()
+    }
+
+    fn deserialize_byte_matrix_hex<'de, D>(deserializer: D) -> Result<Vec<Vec<u8>>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let vec_s: Vec<String> = Vec::deserialize(deserializer)?;
+        vec_s
+            .into_iter()
+            .map(|s| Vec::<u8>::from_hex(&s).map_err(serde::de::Error::custom))
+            .collect()
+    }
+
+    #[allow(dead_code)]
```

Comment:
```suggestion
    #[expect(dead_code)]
```

If this thing is used, then this `expect` lint would fail.
I prefer that to `allow` which flies under the radar.

Reply:
`#[expect(dead_code)]` complains that the actually dead code is supposedly not dead. I left `#[allow(dead_code)]` for now.

```
warning: this lint expectation is unfulfilled
   --> bitvm/src/signatures/public.rs:637:14
    |
637 |     #[expect(dead_code)]
    |              ^^^^^^^^^
    |
    = note: `#[warn(unfulfilled_lint_expectations)]` on by default
```